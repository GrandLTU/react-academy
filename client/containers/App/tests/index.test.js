import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { AppDisconnected as App } from '../index';


describe('App', () => {

    let wrapper = {};

    beforeEach(() => {
        const props = {
            main: {
                value: 5
            }
        };
        wrapper = shallow(<App {...props} />);
    });

    it ('not crash', () => {
        expect(wrapper);
    });

    it('match a snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('match a snapshot 2', () => {
        const props = {
            main: () => {}
        };
        wrapper = shallow(<App {...props} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('clear state', () => {
        wrapper.instance().onRemoveAllCards();

        expect(wrapper.state()).toEqual({ cards: [] });
    });

    it('have initial state', () => {
        expect(wrapper.state()).toEqual({ cards: [ { value: 1 } ] });
    });

});


