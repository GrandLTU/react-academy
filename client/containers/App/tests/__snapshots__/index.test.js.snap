// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App match a snapshot 1`] = `
<div
  className="app-container"
>
  <div
    className="play-area"
  >
    <div
      className="play-area-cards"
    >
      <div
        className="play-area-cards__deck"
      >
        <Card
          deck={true}
          value={52}
        />
      </div>
      <div
        className="play-area-cards__items"
      >
        5
      </div>
    </div>
    <div
      className="play-area__actions"
    >
      <Button
        click={[Function]}
        title="Add card"
      />
      <Button
        click={[Function]}
        title="Remove cards"
      />
    </div>
  </div>
</div>
`;

exports[`App match a snapshot 2 1`] = `
<div
  className="app-container"
>
  <div
    className="play-area"
  >
    <div
      className="play-area-cards"
    >
      <div
        className="play-area-cards__deck"
      >
        <Card
          deck={true}
          value={52}
        />
      </div>
      <div
        className="play-area-cards__items"
      />
    </div>
    <div
      className="play-area__actions"
    >
      <Button
        click={[Function]}
        title="Add card"
      />
      <Button
        click={[Function]}
        title="Remove cards"
      />
    </div>
  </div>
</div>
`;
